frontend/src/concepts/pipelines/topology/parseUtils.ts

```diff
@@ -44,33 +44,47 @@ export const parseComponentsForArtifactRelationship = (
   );
 
 export type TaskArtifactMap = {
-  [taskName: string]: { outputArtifactKey: string; artifactId: string }[] | undefined;
+  [taskName: string]: { artifactNodeId: string }[] | undefined;
 };
-export const parseTasksForArtifactRelationship = (tasks: DAG['tasks']): TaskArtifactMap =>
-  Object.values(tasks).reduce<TaskArtifactMap>(
-    (map, taskValue) =>
-      Object.entries(taskValue.inputs?.artifacts ?? {}).reduce(
-        (artifactItems, [artifactId, value]) => {
-          const { producerTask: taskId, outputArtifactKey } = value.taskOutputArtifact || {};
-          if (!taskId || !outputArtifactKey) {
-            // eslint-disable-next-line no-console
-            console.warn('Issue constructing artifact node', value);
-            return artifactItems;
-          }
-
+export const parseTasksForArtifactRelationship = (
+  groupId: string | undefined,
+  tasks: DAG['tasks'],
+): TaskArtifactMap =>
+  Object.entries(tasks).reduce<TaskArtifactMap>(
+    (map, [taskId, taskValue]) =>
+      Object.entries(taskValue.inputs?.artifacts ?? {}).reduce((artifactItems, [, value]) => {
+        // artifact without inputs
+        if (value.componentInputArtifact) {
           return {
             ...artifactItems,
             [taskId]: [
               ...(artifactItems[taskId] ?? []),
               {
-                outputArtifactKey,
-                artifactId,
+                artifactNodeId: idForTaskArtifact(groupId, '', value.componentInputArtifact),
               },
             ],
           };
-        },
-        map,
-      ),
+        }
+
+        // else, artifacts with inputs from tasks
+        const { producerTask, outputArtifactKey } = value.taskOutputArtifact || {};
+
+        if (!producerTask || !outputArtifactKey) {
+          // eslint-disable-next-line no-console
+          console.warn('Issue constructing artifact node', value);
```

Comment:
do we need this still?

Reply:
might not be your code but we should remove it

Reply:
Yeah, I think we still want this. When the artifact node is from tasks, we want to make sure both values are there to create the edge. Otherwise, we still want to print a warning in the console.

Reply:
We can remove it, it won't hurt anything if you think it's ok not to print any warning, but simply silence the error.

Reply:
we can keep it.